metabase/metabase
# Fix broken popover in Segments form

Closes https://github.com/metabase/metabase/issues/46714

### Description

Fixes #46714 by ensuring that the segment date picker isn't prematurely closed when trying to access the `Starting from` or `include (period)` options in the relative datepicker. The issue occurs because the segment filter popover (located within the `GuiQueryEditor` component) uses the old `TippyPopover` component, which saw any clicks on the relative date picker's options popover as "outside" its catchment area. Therefore, it would close the popover and the user would lose their filter options.

I've also updated the `OperatorSelector` component to use the Mantine Select component, which also had the problem of being hidden behind the main filter popover. I've updated the component and converted it to a smaller TS component. No functionality should have changed.


### How to verify

_**Relative Date Picker**_
1. Go to /admin/datamodel/segment/create
2. Choose "Orders"
3. Filter by "Created At"
4. Choose "Relative dates..."
5. Choose either "Previous" or "Next" (it doesn't matter)
6. Click on the "ellipsis" menu and try choosing "Starting from"
7. The popover should not disappear after clicking on "Starting from" (or "include today"). The filter should be applied as normal


_**Operator Selector**_
1. Go to /admin/datamodel/segment/create
2. Choose "Orders"
3. Filter by "Total"
4. Click on the operator selector (the popover with `Equal to`)
5. See that the menu opens in front of the popover

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist
